### PR TITLE
Add a --short flag to kubectl config view

### PR DESCRIPTION
**Pull request by j3ffml**
Defaults to false, when enabled, redacts raw data for a shorter,
human-readable view. Useful for inspecting files that have embeded
data. We may actually want to make the default be --short=true, and
disable it for --flatten.

cc @deads2k, @satnam6502

### CODE REVIEW

File: pkg/kubectl/cmd/config/view.go [hunk 1]
```diff
@@ -83,6 +85,7 @@ func NewCmdConfigView(out io.Writer, ConfigAccess ConfigAccess) *cobra.Command {
 
 	options.Merge.Default(true)
 	cmd.Flags().Var(&options.Merge, "merge", "merge together the full hierarchy of kubeconfig files")
+	cmd.Flags().BoolVar(&options.Short, "short", false, "redact raw data for a shorter, human-readable view")
```
Comment (deads2k) [inline pkg/kubectl/cmd/config/view.go]: redact raw **byte** data?

**Comment by deads2k** [comment]
Optional nit, the code lgtm. 

@liggitt  Any thoughts on the defaulting.  It'd be nice to be able to read the output by default, but changing the default value of `short` based on the value of other flags might be confusing.

File: pkg/kubectl/cmd/config/view.go [hunk 2]
```diff
@@ -106,6 +109,11 @@ func (o ViewOptions) Run(out io.Writer, printer kubectl.ResourcePrinter) error {
 		}
 	}
 
+	if o.Short {
+		fmt.Printf("short\n")
```
Comment (liggitt) [inline pkg/kubectl/cmd/config/view.go]: why the extra output? won't this mess up anything expecting structured output, or anything using templated output?

Comment (deads2k) [inline pkg/kubectl/cmd/config/view.go]: It's intentional.  For templates, you won't it shortened.  If you're shortening, then the output is no longer a useable kubeconfig, so you don't want it to parse.

File: pkg/client/clientcmd/api/helpers_test.go [hunk 3]
```diff
@@ -202,3 +205,99 @@ func TestFlattenSuccess(t *testing.T) {
...
+	// users:
+	//   red-user:
+	//     LocationOfOrigin: ""
+	//     client-certificate-data: REDACTED
+	//     client-key-data: REDACTED
+	//     token: red-token
+}
+
+func TestShortenSuccess(t *testing.T) {
+	return
```
Comment (satnam6502) [inline pkg/client/clientcmd/api/helpers_test.go]: Why does this function start with `return` ?

**Comment by satnam6502** [comment]
Rather than having `--short` which defaults to true, how about `--full` or `--raw` or `--authinfo` which defaults to false? I fell that `short` is not quite the right words here since what you want to do is to express the idea of just an abstraction or sub-set of the original information being passed through.

**Comment by deads2k** [comment]
You could try going one step further and actually having `--redacted` that substitutes the certs and replaces the tokens and passwords.  It would be removing two non-secrets as well, but would make it easy for people to shared troublesome kubeconfigs for help.

File: pkg/kubectl/cmd/config/view.go (hunk 2 above)
Comment (jlowdermilk) [inline pkg/kubectl/cmd/config/view.go]: Actually it wasn't intentional, just debugging cruft. If we want to make `--short` incompatible with `-o template`, that should be done explicitly.

File: pkg/client/clientcmd/api/helpers_test.go (hunk 3 above)
Comment (jlowdermilk) [inline pkg/client/clientcmd/api/helpers_test.go]: Leftover from testing that I forgot to remove.

**Comment by jlowdermilk** [comment]
@deads2k, @liggitt what do you think of Satnam's idea of using `--raw` for full output, and redacting byte data by default? The other flags (`--flatten`, `--minify`, `--output`) would still output raw byte data, and we can fail the command if users specify incompatible flags (e.g. both `--raw=false` and `--flatten`).

**Comment by jlowdermilk** [comment]
@deads2k, PTAL
Changed flag name to `--raw`, enabled by default, except when `--flatten` is true. Let me know if you have any objections.

File: pkg/client/clientcmd/api/helpers_test.go [hunk 4]
```diff
@@ -202,3 +205,98 @@ func TestFlattenSuccess(t *testing.T) {
...
+	defer os.Remove(keyFile.Name())
+	caFile, _ := ioutil.TempFile("", "")
+	defer os.Remove(caFile.Name())
+
+	certData := "cert"
+	keyData := "key"
+	caData := "ca"
+
+	config := newMergedConfig(certFile.Name(), certData, keyFile.Name(), keyData, caFile.Name(), caData, nil)
+	//config.CurrentContext = "shaker-context"
```
Comment (deads2k) [inline pkg/client/clientcmd/api/helpers_test.go]: delete

**Comment by deads2k** [comment]
lgtm

File: pkg/client/clientcmd/api/helpers_test.go (hunk 4 above)
Comment (jlowdermilk) [inline pkg/client/clientcmd/api/helpers_test.go]: done

**Comment by satnam6502** [comment]
Re-running Shippable.

**Comment by jlowdermilk** [comment]
Can haz merge?

**Pull request merged**